db/db_utils.go

```diff
@@ -54,6 +59,19 @@ func (s *shimFactoryImpl) GetSession() (db.Session, db.Database, error) {
 	return session, session.DB(s.db), nil
 }
 
+func (s *shimFactoryImpl) GetContextSession(ctx context.Context) (db.Session, db.Database, error) {
+	if s.env == nil {
+		return nil, nil, errors.New("undefined environment")
+	}
+
+	session := s.env.ContextSession(ctx)
+	if session == nil {
+		return nil, nil, errors.New("session is not defined")
```

Comment:
should we error "context session is not defined" to distinguish this error from the above function?